File: part1/src/Movies.java
import java.util.List;

public class Movies {

    public static List<Movie> movies() {
        return List.of(
            new Movie("Star Wars", 1977),
            new Movie("Star Trek: the Motion Picture", 1979),
            new Movie("Blade Runner", 1982),
            new Movie("The Matrix", 1999),
            new Movie("Alien", 1979),
            new Movie("Inception", 2010),
            new Movie("Interstellar", 2014),
            new Movie("The Terminator", 1984),
            new Movie("2001: A Space Odyssey", 1968),
            new Movie("E.T. the Extra-Terrestrial", 1982),
            new Movie("Jurassic Park", 1993),
            new Movie("Avatar", 2009)
        );
    }

    public static void main(String[] args) {
        List<Movie> startingMovies = movies();

        /* Write some Java code to do the following:
                1) Filter startingMovies so that it contains only movies from 1980 or later
                   (That is, make a new ArrayList of movies and keep only those from 1980+)

                2) Sort the resulting list alphabetically

                3) Print each movie in the resulting list in the form "(year) Title", e.g. "(1977) Star Wars"
         */

        // 1) Filter startingMovies so that it contains only movies from 1980 or later
        // (startingMovies is immutable, so make a new ArrayList and retain only those movies from 1980 or later)





        // 2) Sort the resulting list alphabetically
        // The easiest way to do this is something like
        //     movies.sort(Comparator.comparing(movie -> movie.title()))





        // 3) Print each movie in the resulting list in the form "(year) Title", e.g. "(1977) Star Wars"





    }
}

File: part2/src/Movies.scala
object Movies {

  // Equivalent to Movies.movies() from part1, except it turns an array instead of a list
  // (Arrays are more useful in Scala than in Java for *reasons*)
  def movies(): Array[Movie] = Array(
    Movie("Star Wars", 1977),
    Movie("Star Trek: the Motion Picture", 1979),
    Movie("Blade Runner", 1982),
    Movie("The Matrix", 1999),
    Movie("Alien", 1979),
    Movie("Inception", 2010),
    Movie("Interstellar", 2014),
    Movie("The Terminator", 1984),
    Movie("2001: A Space Odyssey", 1968),
    Movie("E.T. the Extra-Terrestrial", 1982),
    Movie("Jurassic Park", 1993),
    Movie("Avatar", 2009)
  )

  def main(args: Array[String]): Unit =
    () // replace with code
}
